### PR TITLE
[nrf noup] boot/zephyr/boards: nRF54l15pdk ext flash cfg

### PR DESCRIPTION
**Pull request by nvlsianpu**
Added configuration which allows to build MCUboot for nrf54l15pdk_nrf54l15_cpuapp with external flash used for the secondary slot.

### CODE REVIEW

**Comment by gchwier** [comment]
you don't need to do any changes in mcuboot repo. You can put those configuration files in `sdk-zephyr/samples/subsys/mgmt/mcumgr/smp_svr/child_image/mcuboot/boards`:
nrf54l15pdk_ext_flash.conf -> nrf54l15pdk_nrf54l15_cpuapp.conf
nrf54l15pdk_ext_flash.overlay -> nrf54l15pdk_nrf54l15_cpuapp.overlay

find some details about multi-image builds here: https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/config_and_build/multi_image.html#permanent-configuration-changes-to-child-images

File: boot/zephyr/boards/nrf54l15pdk_nrf54l15_cpuapp_ext_flash.overlay [hunk 1]
```diff
@@ -0,0 +1,10 @@
+/ {
+	chosen {
+		nordic,pm-ext-flash = &mx25r64;
+	};
+};
+
+
+&mx25r64 {
+	status = "okay";
+}
```
Comment (gchwier) [inline boot/zephyr/boards/nrf54l15pdk_nrf54l15_cpuapp_ext_flash.overlay:10]:
```suggestion
};
```

Comment (nvlsianpu) [reply boot/zephyr/boards/nrf54l15pdk_nrf54l15_cpuapp_ext_flash.overlay]: fixed

File: boot/zephyr/boards/nrf54l15pdk_nrf54l15_cpuapp_ext_flash.conf [hunk 2]
```diff
@@ -0,0 +1,8 @@
+CONFIG_MULTITHREADING=y
+CONFIG_SPI=y
+CONFIG_SPI_NOR=y
+CONFIG_FLASH=y
+CONFIG_PM_PARTITION_SIZE_MCUBOOT=0x14000
```
Comment (de-nordic) [inline boot/zephyr/boards/nrf54l15pdk_nrf54l15_cpuapp_ext_flash.conf:5]: This implies usage of Partition Manager, is that correct?

Comment (nvlsianpu) [reply boot/zephyr/boards/nrf54l15pdk_nrf54l15_cpuapp_ext_flash.conf:5]: yes

**Review by de-nordic — APPROVED** [review]

**Review by gchwier — APPROVED** [review]

**Comment by nvlsianpu** [comment]
@de-nordic Need to be reapplied

**Pull request merged**